src/Microsoft.OData.Core/TaskUtils.cs

```diff
@@ -55,6 +65,38 @@ internal static Task CompletedTask
             }
         }
 
+        /// <summary>
+        /// Returns a completed task with the result set to true.
+        /// </summary>
+        internal static Task<bool> TrueResultTask

```

Comment:
Seems like the built-in `Task.FromResult` already caches completed tasks for boolean values (see https://source.dot.net/#System.Private.CoreLib/Task.cs,5198). Maybe this might be redundant?